artemis-protocols/artemis-mqtt-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/mqtt/MQTTConnectionManager.java

```diff
@@ -176,33 +174,27 @@ ServerSessionImpl createServerSession(String username, String password, String v
       return (ServerSessionImpl) serverSession;
    }
 
-   void disconnect(boolean failure) {
+   synchronized void disconnect(boolean failure) {
       if (session == null || session.getStopped()) {
          return;
       }
 
-      synchronized (session.getState()) {
```

Comment:
This seems fine in terms of protecting the connect+disconnect calls similar to before...but, are there other areas synchronising on and accessing or manipulating the session state that might no longer be protected as before that still need to be?

Reply:
I've looked through it carefully and I don't see any potential problems. The relevant changes to the state are performed by `connect` and `disconnect` which are themselves now use `synchronized`.